spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -174,6 +174,27 @@ public void testRewriteDataFilesWithFilterOnPartitionTable() {
     assertEquals("Data after compaction should not change", expectedRecords, actualRecords);
   }
 
+  @Test
+  public void testRewriteDataFilesWithInFilterOnPartitionTable() {
```

Comment:
Lets add in some tests for some of the other possible Filters, I think it would even be ok to do all of them in one test but it may make sense to separate them out for readability. 

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

Reply:
Do we need check for all data type aswell ? Then it may become 15 (type of filters) * 10 (type of data type) scenarios

Reply:
I would just put all filters in the same query, or maybe group into 3 queries, I don't think data types is important here. At least from my perspective the difficult here is in resolving the operator portion, not the literal